### PR TITLE
Handle x-kubernetes-preserve-unknown-fields in CRD for Kubernetes Resources object.

### PR DESCRIPTION
**Pull request by Gartium**
### Type of change

- Enhancement / new feature

### Description

Generate correct schema for x-kubernetes-int-or-string
Related issue https://github.com/strimzi/strimzi-kafka-operator/issues/9851

### Checklist

- [v] Make sure all tests pass

### CODE REVIEW

File: crd-generator/src/test/java/io/strimzi/crdgenerator/ExampleCrd.java [hunk 1]
```diff
@@ -54,10 +55,10 @@
         }
     )
 )
-@OneOf({@OneOf.Alternative(@OneOf.Alternative.Property("either")), @OneOf.Alternative(@OneOf.Alternative.Property("or")), @OneOf.Alternative({@OneOf.Alternative.Property("mapStringString"), @OneOf.Alternative.Property("mapStringObject")})})
+@OneOf({@OneOf.Alternative(@OneOf.Alternative.Property("either")), @OneOf.Alternative(@OneOf.Alternative.Property("or")), @OneOf.Alternative({@OneOf.Alternative.Property("mapStringString"), @OneOf.Alternative.Property("mapStringQuantity"), @OneOf.Alternative.Property("mapStringObject")})})
```
Comment (scholzj) [inline crd-generator/src/test/java/io/strimzi/crdgenerator/ExampleCrd.java:58]: Why do you tie it into the alternative fields? I do not understand the relationship of it.

File: crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java [hunk 2]
```diff
@@ -885,9 +890,16 @@ private ObjectNode buildArraySchema(ApiVersion crApiVersion, Property property,
...
 
-    private boolean isStringStringMap(PropertyType propertyType) {
+    /**
+     * Utility method to check if Map key-value pair match specific types.
+     * @param propertyType property to check
+     * @param keyType key Class
+     * @param valueType value Class
+     * @return true if key-value types are equal to specified types, false otherwise.
+     */
+    private boolean isSameTypesMap(PropertyType propertyType, Class<?> keyType, Class<?> valueType) {
```
Comment (scholzj) [inline crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java:900]: I don't think the name makes sense as the types are not the same. You have different types for key and value. So you should probably find a better name? Maybe `isMapWithTypes`?

**Review by scholzj — COMMENTED** [review]
I left some nits. It might be also useful to add the resource request / limit to one of the `api` module IT test - e.g. for the Kafka CR or something and test that it is properly envoded / decoded.

**Comment by Gartium** [comment]
Addressed comments

File: crd-generator/src/test/java/io/strimzi/crdgenerator/ExampleCrd.java (hunk 1 above)
Comment (Gartium) [reply crd-generator/src/test/java/io/strimzi/crdgenerator/ExampleCrd.java]: Removed, it from alternatives. Wasn't sure if it should be part of it.

File: crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java (hunk 2 above)
Comment (Gartium) [reply crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java]: renamed to `isMapOfTypes`

**Review by scholzj — APPROVED** [review]
LGTM. Thanks. Although I still think this would be a good idea:

> It might be also useful to add the resource request / limit to one of the api module IT test - e.g. for the Kafka CR or something and test that it is properly envoded / decoded.

**Review by ppatierno — APPROVED** [review]

**Pull request merged**